macOS/LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Authentication/DataBrokerProtectionAuthenticationManaging.swift

```diff
@@ -36,7 +36,15 @@ public final class DataBrokerProtectionAuthenticationManager: DataBrokerProtecti
     private let subscriptionManager: DataBrokerProtectionSubscriptionManaging
 
     public var isUserAuthenticated: Bool {
-        subscriptionManager.isUserAuthenticated
+        var token: String?
+        // extremely ugly hack, will be removed as soon auth v1 is removed
+        let semaphore = DispatchSemaphore(value: 0)
+        Task {
+            token = await accessToken()
+            semaphore.signal()
+        }
+        semaphore.wait()
+        return token != nil
```

Comment:
I think this hack concerns me a little bit. How long is it likely to be before we can remove it?

Reply:
As soon AuthV2 is released and we remove AuthV1 code, this is the task: https://app.asana.com/0/72649045549333/1209279739041315 by current estimation this should be done around the 14/04